dotnet/src/Connectors/Connectors.AzureOpenAI/AddHeaderRequestPolicy.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Microsoft.SemanticKernel.Connectors.AzureOpenAI;
+
+/// <summary>
+/// Helper class to inject headers into Azure SDK HTTP pipeline
+/// </summary>
+internal sealed class AddHeaderRequestPolicy(string headerName, string headerValue) : HttpPipelineSynchronousPolicy
```

Comment:
Possibly move this to internal utilities also

Reply:
Captured in the TODO list and will be working on in a separate PR.